test/extended/fixtures/test-build.json

```diff
@@ -38,6 +38,7 @@
         "strategy": {
           "type": "Source",
           "sourceStrategy": {
+            "env": [ { "name": "FOO", "value": "test" } ],
```

Comment:
you should add a second env var that is not expected to be overwritten and confirm that it is not changed.